code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/edge/consent/ConsentEdgeResponseHandlingTests.java

```diff
@@ -197,8 +275,9 @@ public void test_EdgeResponse_NoConsentChangeAndSameTimestamp() throws Exception
 		resetTestExpectations();
 
 		// read timestamp from XDM shared state
-		Map<String, String> xdmSharedState = flattenMap(getXDMSharedStateFor(ConsentConstants.EXTENSION_NAME, 1000));
-		String timestamp = xdmSharedState.get("consents.metadata.time");
+		Map<String, Object> xdmSharedState = getXDMSharedStateFor(ConsentConstants.EXTENSION_NAME, 1000);
```

Comment:
Extract this into a helper function 
getConsentTSFromXDMSharedState()

and add null checks and return -1 or any failing value for time in case of nulls.

Reply:
@addb timestamp is a String. Should I return some like " return "Error: xdmSharedState is null;" when it is null?